setup_unit_tests.py

```diff
@@ -0,0 +1,488 @@
+    To be used if a wdl file has a type that miniwdl doesn't support, ex Object
+    """
+    regex_var_types = re.compile(regex_var_types_str)
+
+    with open(wdl_file, "r") as f:
+        wdl_contents = f.read()
+
+    var_types = {}
+    # can't regex this as regex cannot handle nested constructs
+    # this is a little fragile if the entire declaration isn't on the same line
+    # find all var declarations
+    for line in wdl_contents.split("\n"):
+        line = line.strip()
+        if "=" in line:
+            match = re.search(regex_var_types, line)
+            if match is None:
+                # this line is not a var declaration
+                continue
+            name = match.group(2)
+            typ = match.group(1)
+            var_types[name] = typ
+
+    return var_types
+
+
+def wdl_type_to_string(wdl_type: WDL.Type.Base) -> Union[Dict[str, Any], str]:
+    if isinstance(wdl_type, WDL.Type.StructInstance):
+        # this is the only construct that we need to decipher to string form
+        dictified = {}
+        for key, subtype in wdl_type.members.items():
+            dictified[key] = wdl_type_to_string(subtype)
+        return dictified
+    return str(wdl_type)
+
+
+def extract_output_types(wdl_file, fail):
+    """
+    Use miniwdl's parser to extract the output types from a wdl file
+    """
+    if fail:
+        # since we expect this workflow to fail, there should be no outputs
+        return {}
+
+    try:
+        document: WDL.Tree.Document = WDL.load(uri=str(wdl_file))
+    except WDL.Error.InvalidType as e:
+        if "Unknown type Object" in str(e):
+            # one of the tests wants to test if Objects are supported, but miniwdl doesn't support objects
+            # defer to the other more simpler parser
+            return extract_output_types_defer(wdl_file)
+        else:
+            raise RuntimeError("Likely unsupported type. Failed to parse WDL file %s!", wdl_file) from e
+    except Exception as e:
+        raise RuntimeError(f"Failed to parse WDL file {str(wdl_file)}!") from e
+
+    target: Union[WDL.Tree.Workflow, WDL.Tree.Task]
+
+    if document.workflow:
+        target = document.workflow
+    elif len(document.tasks) == 1:
+        target = document.tasks[0]
+    else:
+        raise Exception(
+            "Multiple tasks founds with no workflow!")  # this in theory shouldn't be hit, but the source repo has a suspicious --tasks argument passed to miniwdl sometimes
+
+    var_types = dict()
+    if target.outputs is not None:
+        for output in target.outputs:
+            var_types[output.name] = wdl_type_to_string(output.type)
+
+    return var_types
+
+
+def convert_typed_output_values(output_values: Union[None, str, Dict[str, Any], List[Any]], output_type: WDL.Type.Base,
+                                data_dir: Optional[Path], metadata: Union[List[Any], Dict[str, Any]]):
+    """
+    Get the expected output values with respect to the output types.
```

Comment:
So we're using the types to somehow infer what the values should be, given that we know their types?

Reply:
Yeah, particularly for the File type which we test against an md5sum hash or try to match a regex expression to